includes/admin/load.php

```diff
@@ -224,31 +228,85 @@ function perflab_enqueue_modules_page_scripts() {
 	wp_enqueue_script( 'plugin-install' );
 
 	wp_enqueue_script(
-		'perflab-plugin-management',
-		plugin_dir_url( __FILE__ ) . 'js/perflab-plugin-management.js',
+		'perflab-install-activate-plugins',
+		plugin_dir_url( __FILE__ ) . 'js/perflab-install-activate-plugins.js',
 		array( 'jquery' ),
 		'1.0.0',
 		array(
 			'in_footer' => true,
 			'strategy'  => 'defer',
 		)
 	);
+
+	wp_localize_script(
+		'perflab-install-activate-plugins',
+		'perflab_install_activate_plugins',
+		array(
+			'ajaxurl' => admin_url( 'admin-ajax.php' ),
+			'nonce'   => wp_create_nonce( 'perflab-install-activate-plugins' ),
+		)
+	);
 }
 
+// WordPress AJAX action to handle the button click event.
+add_action( 'wp_ajax_perflab_install_activate_plugins', 'perflab_install_activate_plugins_callback' );
+
 /**
- * Callback function hooked to admin_action_perflab_activate_plugin to handle plugin activation.
+ * Handles the performance plugin install and activation via AJAX.
  *
- * @since 2.8.0
+ * @since n.e.x.t
  */
-function perflab_activate_plugin() {
-	// Do not proceed if plugin query arg is not present.
-	if ( empty( $_GET['plugin'] ) ) {
-		return;
+function perflab_install_activate_plugins_callback() {
+	if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'perflab-install-activate-plugins' ) ) {
```

Comment:
One more check can be added to check if `$_REQUEST['nonce']` exists.

Reply:
Actually, I think `wp_verify_nonce()` can rather be replaced with `check_ajax_referer()`:

```suggestion
	if ( ! check_ajax_referer( 'perflab-install-activate-plugins', 'nonce', false ) ) {
```